routers/api/v1/repo/mirror.go

```diff
@@ -63,3 +74,318 @@ func MirrorSync(ctx *context.APIContext) {
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: page
+	//   in: query
+	//   description: page number of results to return (1-based)
+	//   type: integer
+	// - name: limit
+	//   in: query
+	//   description: page size of results
+	//   type: integer
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/PushMirrorList"
+	//   "400":
+	//     "$ref": "#/responses/error"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+
+	if !setting.Mirror.Enabled {
+		ctx.Error(http.StatusBadRequest, "GetPushMirrorsByRepoID", "Mirror feature is disabled")
+		return
+	}
+
+	repo := ctx.Repo.Repository
+	listOptions := utils.GetListOptions(ctx)
+	// Get All push mirrors of a specific repo
+	pushMirrors, err := repo_model.GetPushMirrorsByRepoID(repo.ID, listOptions)
+	if err != nil {
+		ctx.Error(http.StatusBadRequest, "GetPushMirrorsByRepoID", err)
+		return
+	}
+
+	responsePushMirrors := make([]*api.PushMirror, 0)
+	for _, mirror := range pushMirrors {
+		m, err := convert.ToPushMirror(mirror)
+		if err == nil {
+			responsePushMirrors = append(responsePushMirrors, m)
+		}
+
+	}
+	ctx.SetLinkHeader(len(pushMirrors), listOptions.PageSize)
+	ctx.SetTotalCountHeader(int64(len(pushMirrors)))
+	ctx.JSON(http.StatusOK, responsePushMirrors)
+}
+
+// GetPushMirrorByName get push mirror of a repository by name
+func GetPushMirrorByName(ctx *context.APIContext) {
+	// swagger:operation GET /repos/{owner}/{repo}/push_mirrors/{name} repository repoGetPushMirrorByRemoteName
+	// ---
+	// summary: Get push mirror of the repository by remoteName
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: owner
+	//   in: path
+	//   description: owner of the repo
+	//   type: string
+	//   required: true
+	// - name: repo
+	//   in: path
+	//   description: name of the repo
+	//   type: string
+	//   required: true
+	// - name: name
+	//   in: path
+	//   description: remote name of push mirror
+	//   type: string
+	//   required: true
+	// responses:
+	//   "200":
+	//     "$ref": "#/responses/PushMirror"
+	//   "400":
+	//     "$ref": "#/responses/error"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+
+	if !setting.Mirror.Enabled {
+		ctx.Error(http.StatusBadRequest, "GetPushMirrorByRemoteName", "Mirror feature is disabled")
+		return
+	}
+
+	mirrorName := ctx.Params(":name")
+	// Get push mirror of a specific repo by remoteName
+	pushMirror, err := repo_model.GetPushMirrors(repo_model.PushMirrorOptions{RepoID: ctx.Repo.Repository.ID, RemoteName: mirrorName})
+	if err != nil {
+		ctx.Error(http.StatusBadRequest, "GetPushMirrorByRemoteName", err)
+		return
+	}
+	m, err := convert.ToPushMirror(pushMirror)
+	if err != nil {
+		ctx.Error(http.StatusBadRequest, "GetPushMirrorByRemoteName", err.Error())
```

Comment:
This should be InternalServerError.
```suggestion
		ctx.ServerError("GetPushMirrorByRemoteName", err)
```